### PR TITLE
11 SubpopulationController, SubstudyController, SurveyController

### PR DESCRIPTION
**Pull request by alxdarksage**

### CODE REVIEW

File: src/test/java/org/sagebionetworks/bridge/spring/controllers/SurveyControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,635 @@
...
+        
+        StatusMessage result = controller.deleteSurvey(SURVEY_GUID, CREATED_ON.toString(), true);
+        assertEquals(result, SurveyController.DELETED_MSG);
+        
+        verify(mockSurveyService).getSurvey(TEST_STUDY, KEYS, false, false);
+        verify(mockSurveyService).deleteSurvey(TEST_STUDY, survey);
+        verifyNoMoreInteractions(mockSurveyService);
+    }
+
+    public void physicalDeleteAllowedForAdmin() throws Exception {
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/spring/controllers/SurveyControllerTest.java:414]: Minor: Test annotation.

File: src/test/java/org/sagebionetworks/bridge/spring/controllers/SurveyControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,635 @@
...
+    @Test
+    public void versionSurveyInvalidatesCache() throws Exception {
+        assertCacheIsCleared((guid, dateString) -> controller.versionSurvey(guid, dateString));
+    }
+    
+    @Test
+    public void updateSurveyInvalidatesCache() throws Exception {
+        assertCacheIsCleared((guid, dateString) -> controller.updateSurvey(guid, dateString));
+    }
+
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/spring/controllers/SurveyControllerTest.java:576]: Missing test publishSurveyInvalidatesCache()?

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]

File: src/test/java/org/sagebionetworks/bridge/spring/controllers/SurveyControllerTest.java (hunk 1 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/spring/controllers/SurveyControllerTest.java:414]: Added

File: src/test/java/org/sagebionetworks/bridge/spring/controllers/SurveyControllerTest.java (hunk 2 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/spring/controllers/SurveyControllerTest.java:576]: Yes, restored with a boolean third argument.

**Review by DwayneJengSage — APPROVED** [review]

**Pull request merged**